newIDE/app/src/AssetStore/AssetPackDialog.js

```diff
@@ -0,0 +1,210 @@
+          });
+        });
+
+        await resourcesFetcher.ensureResourcesAreFetched(project);
+
+        setIsAssetPackBeingInstalled(false);
+        onAssetPackAdded();
+      } catch (error) {
+        setIsAssetPackBeingInstalled(false);
+        console.error('Error while installing the asset pack', error);
+        showErrorBox({
+          message:
+            'There was an error while installing the asset pack. Verify your internet connection or try again later.',
+          rawError: error,
+          errorId: 'install-asset-pack-error',
+        });
+      }
+    },
+    [
+      resourcesFetcher,
+      eventsFunctionsExtensionsState,
+      project,
+      objectsContainer,
+      events,
+      onObjectAddedFromAsset,
+      onAssetPackAdded,
+    ]
+  );
+
+  return (
+    <Dialog
+      title={<Trans>{assetPack.name}</Trans>}
+      open
+      cannotBeDismissed
+      actions={[
+        <FlatButton
+          key="cancel"
+          label={<Trans>Cancel</Trans>}
+          disabled={isAssetPackBeingInstalled}
+          onClick={onClose}
+        />,
+        // Loading.
+        isAssetPackBeingInstalled ? (
+          <FlatButton
+            key="loading"
+            label={<Trans>Please wait...</Trans>}
+            disabled
+            onClick={() => {}}
+          />
+        ) : // All assets already installed.
+        allAssetsInstalled ? (
+          <RaisedButton
+            key="install-again"
+            label={<Trans>Install again</Trans>}
+            primary={false}
+            onClick={() => onInstallAssetPack(assetShortHeaders)}
+          />
+        ) : // No assets installed.
+        noAssetsInstalled ? (
+          <RaisedButton
+            key="continue"
+            label={<Trans>Continue</Trans>}
+            primary
+            onClick={() => onInstallAssetPack(assetShortHeaders)}
+          />
+        ) : (
+          // Some assets installed.
+          <RaisedButtonWithSplitMenu
+            label={<Trans>Install the missing assets</Trans>}
+            key="install-missing"
+            primary
+            onClick={() => {
+              onInstallAssetPack(missingAssetShortHeaders);
+            }}
+            buildMenuTemplate={i18n => [
+              {
+                label: i18n._(t`Install all the assets`),
+                click: () => onInstallAssetPack(assetShortHeaders),
+              },
+            ]}
+          />
+        ),
+      ]}
+      onApply={() => {
+        noAssetsInstalled
+          ? onInstallAssetPack(assetShortHeaders)
+          : onInstallAssetPack(missingAssetShortHeaders);
+      }}
+    >
+      <Column>
+        {isAssetPackBeingInstalled ? (
+          <>
+            <Text>
+              <Trans>Installing assets...</Trans>
+            </Text>
+            <Line expand>
+              <LinearProgress style={styles.linearProgress} />
+            </Line>
+          </>
+        ) : (
+          <Text>
+            {allAssetsInstalled ? (
```

Comment:
It can be hard to understand which text comes with which button since they are located far from each other.

You could have a function `getDialogContent` in the component, that returns a tuple (textOrComponent, button) so that one is sure that all the possibilities are well covered.